### DISCUSSION TITLE
Setting WGPU backend without using env variables

### DISCUSSION THREAD

**Discussion by ThomasAlban** [bevyengine/bevy#8667] [Q&A] [answered]
I'm having the same issue as mentioned in #8395 but my organisation does not allow me to set environment variables, so is there any other way I can set the WGPU backend? Thanks

**Comment by nicopap** [comment]
Could you provide more info on your setup?

I assume you are on powershell on Windows. Check this stack overflow question: <https://stackoverflow.com/questions/1420719/powershell-setting-an-environment-variable-for-a-single-command-only>. Does  any of the answer helps you? My Windows knowledge is a bit dated, so I can't give you more precise details.

But setting environment variables should be local to a shell session (I know, it's counter-intuitive), which means there shouldn't be extra permissions needed to set them.

**Comment by JMS55** [comment, marked as answer]
You can use WgpuSettings::backends https://docs.rs/bevy/latest/bevy/render/settings/struct.WgpuSettings.html#structfield.backends

**Comment by hymm** [comment]
You can set it using `WgpuSettings.backends` on the RenderPlugin. https://docs.rs/bevy/latest/bevy/render/settings/struct.WgpuSettings.html
